File: TreasurersApp/Models/ContributorEmailAddress.cs
namespace TreasurersApp.Models
{
    using Newtonsoft.Json;

    public class ContributorEmailAddress
    {
        [JsonProperty("id")]
        public int ContributorEmailAddressId { get; set; } // ContributorEmailAddressID (Primary key)

        [JsonProperty("contributorId")]
        public int ContributorId { get; set; } // ContributorID

        [JsonProperty("emailAddressId")]
        public int EmailAddressId { get; set; } // EmailAddressID

        [JsonProperty("preferred")]
        public bool Preferred { get; set; } // Preferred

        [JsonProperty("createdBy")]
        public System.Guid CreatedBy { get; set; } // CreatedBy

        [JsonProperty("createdDate")]
        public System.DateTime CreatedDate { get; set; } // CreatedDate

        [JsonProperty("lastModifiedBy")]
        public System.Guid LastModifiedBy { get; set; } // LastModifiedBy

        [JsonProperty("lastModifiedDate")]
        public System.DateTime LastModifiedDate { get; set; } // LastModifiedDate

        // Foreign keys

        /// <summary>
        /// Parent Contributor pointed by [ContributorEmailAddress].([ContributorId]) (FK_ContributorEmailAddress_Contributor)
        /// </summary>
        public Contributor Contributor { get; set; } // FK_ContributorEmailAddress_Contributor

        /// <summary>
        /// Parent EmailAddress pointed by [ContributorEmailAddress].([EmailAddressId]) (FK_ContributorEmailAddress_EmailAddress)
        /// </summary>
        public EmailAddress EmailAddress { get; set; } // FK_ContributorEmailAddress_EmailAddress
    }

}
// </auto-generated>

File: TreasurersApp/ClientApp/src/app/models/ContributorEmailAddress.ts
import { EmailType } from './EmailType';
import { EmailAddress } from './EmailAddress';

export class ContributorEmailAddress {
  id: number = null;
  contributorId: number = null;
  emailAddressId: number = null;
  emailTypeId: number = null;
  preferred: boolean = null;
  createdBy: string = null;
  createdDate: Date = null;
  lastModifiedBy: string = null;
  lastModifiedDate: Date = null;

  email: EmailAddress = null;
  emailType: EmailType = null;

  constructor(
    id: number,
    contributorId: number,
    emailAddressId: number = null,
    emailTypeId: number = null,
    preferred: boolean = null,
    createdBy: string = null,
    createdDate: Date = null,
    lastModifiedBy: string = null,
    lastModifiedDate: Date) {
    this.id = id;
    this.contributorId = contributorId;
    this.emailAddressId = emailAddressId;
    this.emailTypeId = emailTypeId,
    this.preferred = preferred;
    this.createdBy = createdBy;
    this.createdDate = createdDate;
    this.lastModifiedBy = lastModifiedBy;
    this.lastModifiedDate = lastModifiedDate;

 }
}
